gnd/src/main/java/com/google/android/gnd/persistence/local/room/RoomLocalDataStore.java

```diff
@@ -401,17 +423,40 @@ private Completable apply(FeatureMutation mutation) throws LocalDataStoreExcepti
       case CREATE:
         return getUser(mutation.getUserId())
             .flatMapCompletable(user -> insertOrUpdateFeature(mutation, user));
+      case DELETE:
+        return featureDao
+            .findById(mutation.getFeatureId())
+            .flatMapCompletable(entity -> markFeatureForDeletion(entity, mutation))
+            .subscribeOn(schedulers.io());
       default:
         throw LocalDataStoreException.unknownMutationType(mutation.getType());
     }
   }
 
+  private CompletableSource markFeatureForDeletion(FeatureEntity entity, FeatureMutation mutation) {
+    return featureDao
+        .update(entity.toBuilder().setState(EntityState.DELETED).build())
+        .doOnSubscribe(__ -> Timber.d("Marking feature as deleted : %s", mutation))
+        .ignoreElement()
+        .subscribeOn(schedulers.io());
```

Comment:
No need to change the `subscribeOn` thread inside private methods; it only needs to be done in the main entry point of each operation, the setting propagates to the rest of the stream.